### ISSUE TITLE
Vagrant: make git-submodule fails

### ISSUE THREAD

**Issue by atrnh** [qmk/qmk_firmware#5083]
**Describe bug**
`make git-submodule` fails with default installation of Vagrant

**System Information**
 - Keyboard:  
   - Revision (if applicable):  
 - Operating System: macOS w/ Vagrant
 - avr-gcc version: 4.9.22

 - arm gcc version: 4.9.33

 - QMK Firmware version: 0.6.222

 - Any keyboard related software installed? 
   - [ ] Auto Hot Key
   - [ ] Karabiner
   - [ ] Other

**Additional context**
Steps to reproduce:
- [Follow instructions to install/run Vagrant](https://docs.qmk.fm/#/getting_started_vagrant)
- `vagrant up && vagrant ssh`
- `cd /vagrant`
- `make git-submodule`

**Error log**
```
usage: git submodule [--quiet] add [-b <branch>] [-f|--force] [--name <name>] [--reference <repository>] [--] <repository> [<path>]
   or: git submodule [--quiet] status [--cached] [--recursive] [--] [<path>...]
   or: git submodule [--quiet] init [--] [<path>...]
   or: git submodule [--quiet] deinit [-f|--force] [--] <path>...
   or: git submodule [--quiet] update [--init] [--remote] [-N|--no-fetch] [-f|--force] [--checkout|--merge|--rebase] [--reference <repository>] [--recursive] [--] [<path>...]
   or: git submodule [--quiet] summary [--cached|--files] [--summary-limit <n>] [commit] [--] [<path>...]
   or: git submodule [--quiet] foreach [--recursive] <command>
   or: git submodule [--quiet] sync [--recursive] [--] [<path>...]
Makefile:588: recipe for target 'git-submodule' failed
```

**Git version:** 2.7.4

**To fix**
```
git submodule sync --recursive && git submodule update --init --recursive
```

`--progress` flag fails for some reason 🤷‍♀️

**Comment by zvecr** [comment]
Version of git on ubuntu 16.04 is old, #4905 seems to fix this as debian 9 ships a newer git (version 2.11).

**Comment by atrnh** [comment]
Removing `--progress` from `make git-submodule` seems like a better idea.

Switching from Ubuntu to Debian just to use `--progress` is overkill.

**Comment by zvecr** [comment]
I agree with both statements, upgrading base image and the git dependency within make. My comment was more to link the two issues as i had also observed the same problem, and note the potential workaround.

From the man page:
```
--progress
Progress status is reported on the standard error stream by default when it is attached to a terminal, unless -q is specified. This flag forces progress status even if the standard error stream is not directed to a terminal.
```
Not sure if removing the flag will have any unintended side effects.

**Comment by atrnh** [comment]
@zvecr Right on, thanks for the clarifcation!

**Comment by atrnh** [comment]
I assume the lack of verbosity from leaving out `--progress` can make it seem like git is hanging and not doing anything. Only solution I can think of is `echo`-ing some reassuring messages 🤔

**Comment by zvecr** [comment]
Fixed by #4905.
